cartography/intel/azure/sql.py

```diff
@@ -647,169 +633,148 @@ def load_database_details(neo4j_session, details, update_tag):
+    SET point.name = rp.name,
+    point.location = rp.location,
+    point.restoredate = rp.earliest_restore_date,
+    point.restorepointtype = rp.restore_point_type,
+    point.creationdate = rp.restore_point_creation_date
+    WITH point, rp
+    MATCH (d:AzureSQLDatabase{id: rp.database_id})
     MERGE (d)-[r:CONTAINS]->(point)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for point in restore_points:
-        neo4j_session.run(
-            ingest_restore_points,
-            PointId=point['id'],
-            Name=point['name'],
-            Location=point['location'],
-            RestoreDate=point['earliest_restore_date'],
-            RestorePointType=point['restore_point_type'],
-            CreationDate=get_optional_value(point, ['restore_point_creation_date']),
-            DatabaseId=point['database_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_restore_points,
+        restore_points_list=restore_points,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_transparent_data_encryptions(neo4j_session, encryptions_list, update_tag):
+def _load_transparent_data_encryptions(
+        neo4j_session: neo4j.Session, encryptions_list: List[Dict], update_tag: int,
+) -> None:
     """
     Ingest transparent data encryptions into neo4j.
     """
     ingest_data_encryptions = """
-    MERGE (tae:AzureTransparentDataEncryption{id: {TAEId}})
+    UNWIND {transparent_data_encryptions_list} as e
+    MERGE (tae:AzureTransparentDataEncryption{id: e.id})
     ON CREATE SET tae.firstseen = timestamp(), tae.lastupdated = {azure_update_tag}
-    SET tae.name = {Name},
-    tae.location = {Location},
-    tae.status = {Status}
-    WITH tae
-    MATCH (d:AzureDatabase{id: {DatabaseId}})
+    SET tae.name = e.name,
+    tae.location = e.location,
+    tae.status = e.status
+    WITH tae, e
+    MATCH (d:AzureSQLDatabase{id: e.database_id})
     MERGE (d)-[r:CONTAINS]->(tae)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for encryption in encryptions_list:
-        neo4j_session.run(
-            ingest_data_encryptions,
-            TAEId=encryption['id'],
-            Name=encryption['name'],
-            Location=encryption['location'],
-            Status=encryption['status'],
-            DatabaseId=encryption['database_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_data_encryptions,
+        transparent_data_encryptions_list=encryptions_list,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def cleanup_azure_sql_servers(neo4j_session, subscription_id, common_job_parameters):
+def cleanup_azure_sql_servers(
+        neo4j_session: neo4j.Session, subscription_id: str, common_job_parameters: Dict
+) -> None:
     common_job_parameters['AZURE_SUBSCRIPTION_ID'] = subscription_id
```

Comment:
We don't need this assignment. This would be handled in __init__.py file.

Reply:
Done